src/transaction/log_applier.c

```diff
@@ -7043,142 +7043,180 @@ la_print_log_arv_header (const char *database_name, LOG_ARV_HEADER * hdr, bool v
 }
 
 /*
- * la_log_page_check() - test the transaction log
- *   return: void
- *   log_path: log path
- *   page_num: test page number
+ * la_get_applied_log_info() - when applyinfo utility is executed with -a option, this fuction will try to get applied log information.
+ *   return: NO_ERROR or error code
+ *   database_name(in): db name
+ *   log_path(in): real path of log file
+ *   check_replica_info(in): whether replica node execute applyinfo utility
+ *   verbose(in): applyinfo -v option
+ *   applied_final_lsa(in/out): information for delayed applying copied log
+ *
  */
 int
-la_log_page_check (const char *database_name, const char *log_path, INT64 page_num, bool check_applied_info,
-		   bool check_copied_info, bool check_replica_info, bool verbose, LOG_LSA * copied_eof_lsa,
-		   LOG_LSA * copied_append_lsa, LOG_LSA * applied_final_lsa)
+la_get_applied_log_info (const char *database_name, const char *log_path, bool check_replica_info,
+			 bool verbose, LOG_LSA * applied_final_lsa)
 {
   int error = NO_ERROR;
   int res;
-  char *atchar;
-  char active_log_path[PATH_MAX];
   char *replica_time_bound_str;
+  char log_path_buf[PATH_MAX];
 
   assert (database_name != NULL);
   assert (log_path != NULL);
 
-  atchar = (char *) strchr (database_name, '@');
-  if (atchar)
+  LA_HA_APPLY_INFO ha_apply_info;
+  char timebuf[1024];
+
+  if (realpath (log_path, log_path_buf) != NULL)
     {
-      *atchar = '\0';
+      log_path = log_path_buf;
     }
 
-  /* init la_Info */
-  la_init (log_path, 0);
+  la_init_ha_apply_info (&ha_apply_info);
 
-  if (check_applied_info)
+  res = la_get_ha_apply_info (log_path, database_name, &ha_apply_info);
+  if ((res <= 0) || (ha_apply_info.creation_time.date == 0 && ha_apply_info.creation_time.time == 0))
     {
-      LA_HA_APPLY_INFO ha_apply_info;
-      char timebuf[1024];
-
-      la_init_ha_apply_info (&ha_apply_info);
-
-      res = la_get_ha_apply_info (log_path, database_name, &ha_apply_info);
-      if ((res <= 0) || (ha_apply_info.creation_time.date == 0 && ha_apply_info.creation_time.time == 0))
+      if (res < 0)
 	{
 	  error = res;
-
 	  goto check_applied_info_end;
 	}
+      else
+	{
+	  error = ER_GENERIC_ERROR;
```

Comment:
@joohok 
this code is added.
if `la_get_applied_log_info()` returns NO_ERROR, `check_applied_info` is not set to false.
And then "*** Delay in Applying Copied Log ***" is printed in 3404L even if it does not work.

Reply:
why do you use `ER_GENERIC_ERROR` instead of `ER_FAILED`?

Reply:
Because it is actually not failure case. So, I set as `ER_GENERIC_ERROR`

Reply:
Did you see the error message for `ER_GENERIC_ERROR` ? 
it also means internal system failure.

Reply:
Right. I checked it. It is opposite case.

I searched the ER_FAILED. It will be better because there is a case returning ER_FAILED in the `disk_reserve_sectors()`.
(It is similar case that n_sectors == 0)

Thnak you. I will apply it.